src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -405,7 +405,19 @@ public sealed override IEnumerable<Cci.INamespaceTypeDefinition> GetAnonymousTyp
                 return SpecializedCollections.EmptyEnumerable<Cci.INamespaceTypeDefinition>();
             }
 
+#if DEBUG
+            return iterator();
+
+            IEnumerable<Cci.INamespaceTypeDefinition> iterator()
+            {
+                foreach (NamedTypeSymbol type in Compilation.AnonymousTypeManager.GetAllCreatedTemplates())
```

Comment:
Nit: this could just be `return Compilation.AnonymousTypeManager.GetAllCreatedTemplates.Select(t => t.GetCciAdapter());`